### PR TITLE
Feature/web api

### PR DESCRIPTION
**Pull request by Aplicticus**
WebApi Http Put request

### CODE REVIEW

File: Triangle Streaming Server/Controllers/StreamController.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        private StreamQueueManager streamQueueManager;
+
+        public StreamController(StreamQueueManager streamQueueManager)
+        {
+            this.streamQueueManager = streamQueueManager;
+        }
+
+        [HttpGet]
+        [Route("api/Stream/Connect")]
```
Comment (markieo1) [inline Triangle Streaming Server/Controllers/StreamController.cs]: Cant this "api/Stream" not be on the controller? So only the action should be specified.

File: Triangle Streaming Server/Startup.cs [hunk 2]
```diff
@@ -60,6 +58,8 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
 
 			app.MapWebSocketManager("/send", app.ApplicationServices.GetService<VideoStream>());
 			app.MapWebSocketManager("/receive", receiveStream);
-		}
+            
+            app.UseMvc();
```
Comment (markieo1) [inline Triangle Streaming Server/Startup.cs]: Indentation and such

File: Triangle Streaming Server/TriangleStreamingServer.csproj [hunk 3]
```diff
@@ -14,7 +15,12 @@
     <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="1.1.3" />
     <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="1.1.2" />
     <PackageReference Include="Microsoft.AspNetCore.WebSockets.Server" Version="0.1.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="1.1.2" />
```
Comment (markieo1) [inline Triangle Streaming Server/TriangleStreamingServer.csproj]: Why???

File: Triangle Streaming Server/TriangleStreamingServer.csproj [hunk 4]
```diff
@@ -14,7 +15,12 @@
...
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="1.1.2" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer.Design" Version="1.1.2" />
     <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="1.1.2" />
+    <PackageReference Include="Microsoft.VisualStudio.Web.BrowserLink" Version="1.1.2" />
```
Comment (markieo1) [inline Triangle Streaming Server/TriangleStreamingServer.csproj]: Why?

File: Triangle Streaming Server/Views/Shared/Error.cshtml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿@{
```
Comment (markieo1) [inline Triangle Streaming Server/Views/Shared/Error.cshtml]: Why added?

File: Triangle Streaming Server/Views/Shared/Layout.cshtml [hunk 6]
```diff
@@ -0,0 +1,68 @@
+﻿<!DOCTYPE html>
```
Comment (markieo1) [inline Triangle Streaming Server/Views/Shared/Layout.cshtml]: Why?

File: Triangle Streaming Server/Views/_ViewImports.cshtml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+﻿@using TriangleStreamingServer
```
Comment (markieo1) [inline Triangle Streaming Server/Views/_ViewImports.cshtml]: Why?

File: Triangle Streaming Server/Views/_ViewStart.cshtml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+﻿@{
```
Comment (markieo1) [inline Triangle Streaming Server/Views/_ViewStart.cshtml]: Why?

File: Triangle Streaming Server/bower.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
+﻿{
+  "name": "asp.net",
```
Comment (markieo1) [inline Triangle Streaming Server/bower.json]: Why???

File: Triangle Streaming Server/bundleconfig.json [hunk 10]
```diff
@@ -0,0 +1,24 @@
+﻿// Configure bundling and minification for the project.
```
Comment (markieo1) [inline Triangle Streaming Server/bundleconfig.json]: Why?

File: Triangle Streaming Server/packages.config [hunk 11]
```diff
@@ -1,6 +1,14 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="BouncyCastle" version="1.8.1" targetFramework="net452" />
+  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net452" />
+  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net452" />
+  <package id="Microsoft.AspNet.WebApi.Owin" version="5.2.3" targetFramework="net452" />
```
Comment (markieo1) [inline Triangle Streaming Server/packages.config]: Owin is not needed i think??

File: Triangle Streaming Server/wwwroot/css/Site.css [hunk 12]
```diff
@@ -0,0 +1,43 @@
+﻿body {
```
Comment (markieo1) [inline Triangle Streaming Server/wwwroot/css/Site.css]: Why?

File: Triangle Streaming Server/wwwroot/js/Site.js [hunk 13]
```diff
@@ -0,0 +1 @@
+﻿// Write your Javascript code.
```
Comment (markieo1) [inline Triangle Streaming Server/wwwroot/js/Site.js]: Why?

File: Triangle Streaming Server/Controllers/StreamController.cs (hunk 1 above)
Comment (Aplicticus) [inline Triangle Streaming Server/Controllers/StreamController.cs]: Replaced to: [Route("Stream/Connect")]

File: Triangle Streaming Server/Startup.cs (hunk 2 above)
Comment (Aplicticus) [inline Triangle Streaming Server/Startup.cs]: Used tabs instead of spaces

File: Triangle Streaming Server/Views/Shared/Error.cshtml (hunk 5 above)
Comment (Aplicticus) [inline Triangle Streaming Server/Views/Shared/Error.cshtml]: Removed

File: Triangle Streaming Server/Views/Shared/Layout.cshtml (hunk 6 above)
Comment (Aplicticus) [inline Triangle Streaming Server/Views/Shared/Layout.cshtml]: Removed

File: Triangle Streaming Server/Views/_ViewImports.cshtml (hunk 7 above)
Comment (Aplicticus) [inline Triangle Streaming Server/Views/_ViewImports.cshtml]: Removed

File: Triangle Streaming Server/Views/_ViewStart.cshtml (hunk 8 above)
Comment (Aplicticus) [inline Triangle Streaming Server/Views/_ViewStart.cshtml]: Removed

File: Triangle Streaming Server/bower.json (hunk 9 above)
Comment (Aplicticus) [inline Triangle Streaming Server/bower.json]: Removed

File: Triangle Streaming Server/bundleconfig.json (hunk 10 above)
Comment (Aplicticus) [inline Triangle Streaming Server/bundleconfig.json]: Removed

File: Triangle Streaming Server/packages.config (hunk 11 above)
Comment (Aplicticus) [inline Triangle Streaming Server/packages.config]: Removed

File: Triangle Streaming Server/wwwroot/js/Site.js (hunk 13 above)
Comment (Aplicticus) [inline Triangle Streaming Server/wwwroot/js/Site.js]: Removed

File: Triangle Streaming Server/wwwroot/css/Site.css (hunk 12 above)
Comment (Aplicticus) [inline Triangle Streaming Server/wwwroot/css/Site.css]: Removed

File: Triangle Streaming Server/TriangleStreamingServer.csproj (hunk 4 above)
Comment (Aplicticus) [inline Triangle Streaming Server/TriangleStreamingServer.csproj]: Removed

File: Triangle Streaming Server/TriangleStreamingServer.csproj (hunk 3 above)
Comment (Aplicticus) [inline Triangle Streaming Server/TriangleStreamingServer.csproj]: Removed

File: Triangle Streaming Server/Controllers/StreamController.cs [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+        [Route("Send/{id?}")]
+        public async Task<IActionResult> Send(Guid id)
+        {
+            StreamContent sc = new StreamContent(HttpContext.Request.Body);
+            var result = await sc.ReadAsByteArrayAsync();  
+            streamQueueManager.AddToQueue(id.ToString(), result);
+            return Ok();
+        }
+
+        public struct key
```
Comment (markieo1) [inline Triangle Streaming Server/Controllers/StreamController.cs]: Renamed to more logic name

File: Triangle Streaming Server/TriangleStreamingServer.csproj [hunk 15]
```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp1.1</TargetFramework>
+    <PackageTargetFallback>portable-net45+win8</PackageTargetFallback>
```
Comment (markieo1) [inline Triangle Streaming Server/TriangleStreamingServer.csproj]: Why is this added?
